src/core/fetchRequest.js

```diff
@@ -102,6 +102,14 @@ const requestFactory = (url, options) => {
 
             // successful request
             if ((responseCode >= 200 && responseCode < 300) || (response && response.success === true)) {
+                if(options.method === "HEAD"){
+                    const headers = {};
+                    for (const [key, value] of originalResponse.headers.entries()) {
+                        headers[key] = value;
+                    }
+                    return Object.assign(response, {headers});
+                }
```

Comment:
**nitpick:** `entries()` is not supported by [all browsers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries#browser_compatibility) we must support, and it is not polyfilled.

**thought:** Is it actually needed to apply such a loop to copy the object? Wouldn't a shallow copy of the object be enough?

**suggestion:** Consider using a more classical approach, for instance something like:
```suggestion
                if(options.method === "HEAD"){
                    const headers = Object.assign({}, originalResponse.headers);
                    return Object.assign(response, { headers });
                }
```

Or even, if the object does not need to be cloned:
```suggestion
                if(options.method === "HEAD"){
                    const { headers } = originalResponse;
                    return Object.assign(response, { headers });
                }
```

Reply:
Good catch about supporting! I forget that Curr Gen supports wider version of browsers.

Reply:
I created one more approach for headers collection. I've updated PR description with more info about situation.